Examples/Multiplayer/CSharp/Foundation-Unity-Example/Assets/Scripts/HelloWorld.cs

```diff
@@ -0,0 +1,649 @@
+    }
+
+    /// <summary>
+    /// Requests the Foundation layer to Login with email and password.
+    /// </summary>
+    /// <param name="email"> Email of the user. </param>
+    /// <param name="password"> Password of the user. </param>
+    /// <returns> Just the Task object to await.</returns>
+    private async Task LoginAsync(string email, string password)
+    {
+        Debug.Log($"Logging in with Email {email} ...");
+
+        using CspSystems.LoginStateResult loginResult = await userSystem.Login(string.Empty, email, password, true);
+
+        // Cache user ID for later use.
+        userId = loginResult.GetLoginState().UserId;
+        Debug.Log($"Logged in with Email {email}.");
+    }
+
+    /// <summary>
+    /// Requests the Foundation layer to make the user logout.
+    /// Not used for this example as shutting down the Foundation layer will automatically log the user out.
+    /// </summary>
+    /// <returns> Just the Task object to await.</returns>
+    private async Task LogoutAsync()
+    {
+        Debug.Log("Logging out ...");
+        await userSystem.Logout();
+    }
+
+    #endregion
+
+    // Spaces - creating, joining, deleting a space
+
+    #region Spaces
+
+    /// <summary>
+    /// Creates a simple space with the provided name.
+    /// </summary>
+    /// <param name="spaceName">The name of the space</param>
+    private async void CreateSpaceAsync(string spaceName)
+    {
+        Debug.Log($"Creating space: {spaceName} ...");
+
+        var metadata = new Dictionary<string, string>()
+        {
+            { "site", defaultSpaceSite }
+        };
+
+        // Create a public space so other users can join without invite.
+        using CspSystems.SpaceResult spaceResult = await spaceSystem.CreateSpace(spaceName, string.Empty,
+            CspSystems.SpaceAttributes.Public, null, ToFoundationMap(metadata), null);
+
+        createdSpace = spaceResult.GetSpace();
+        Debug.Log($"Created space with name: {createdSpace.Name} and ID: {createdSpace.Id}");
+    }
+
+    /// <summary>
+    /// Enters the space with the given spaceId. If the space was just created, the
+    /// useCreatedSpace is used to enter that space instead.
+    /// </summary>
+    /// <param name="spaceId">The destination space's ID.</param>
+    /// <param name="useCreatedSpace">Whether the user wants to enter the space they just created instead of another specified one via the spaceId parameter.</param>
+    private async void EnterSpaceAsync(string spaceId, bool useCreatedSpace)
+    {
+        string targetSpaceId;
+
+        if (useCreatedSpace)
+        {
+            targetSpaceId = createdSpace.Id;
+        }
+        else
+        {
+            if (string.IsNullOrEmpty(spaceId)) return;
+
+            targetSpaceId = spaceId;
+        }
+
+        using CspSystems.SpaceResult spaceResult = await spaceSystem.GetSpace(targetSpaceId);
```

Comment:
Can we guard against a failure to get the Space. In multiplayer testing with Lyubo on Friday, when Private Spaces were being created, we thought that the non-creator tester was successfully getting the Space due to all the messages subsequently being logged after the call to GetSpace had in fact failed.